### ISSUE TITLE
High Priority - WCAG Level A -Autocomplete. A combobox does not function as expected and is orphaned from its label

### ISSUE THREAD

**Issue by JuliaSang** [w3c/w3c-website#163]
A combobox is present which does not function in the expected manner, and is also orphaned from its label.

**WCAG Reference:**
3.3.2 Labels or Instructions (Level A)
[Understanding Labels or Instructions](https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html) | [How to Meet Labels or Instructions](https://www.w3.org/WAI/WCAG21/quickref/#labels-or-instructions)

1.3.1 Info and Relationships (Level A)
[Understanding Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html) | [How to Meet Info and Relationships](https://www.w3.org/WAI/WCAG21/quickref/#info-and-relationships)

4.1.2 Name, Role, Value (Level A)
[Understanding Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) | [How to Meet Name, Role, Value](https://www.w3.org/WAI/WCAG21/quickref/#name-role-value)

**Issue ID: Autocomplete_01**

URL: https://beta.w3.org/membership/list/ 
Page title: Current Members & Testimonials 

Journey 2, Step 1
<img width="773" alt="autocomplete" src="https://user-images.githubusercontent.com/20167528/159246442-43d7bde3-162a-4170-9963-e9a488eb2e73.png">

The 'Countries/Region' combobox does not function in the expected manner for users that navigate using the keyboard, as although it is possible to type into the form and cycle through the options which are displayed, it is not possible to make a selection using the space bar as would typically be expected, although pressing enter does work, users would expect to be able to make a selection by pressing Space.

In addition to this the combobox presents as being unlabelled to screen reader users due to the labels for the input not correctly referencing the ID of the corresponding component; an aria-labelledby attribute has been assigned as a means of providing a label but this does not reference a corresponding id. As the input is effectively orphaned from its label, it will be difficult for screen reader users to determine its purpose.

Voice activation users were not able to access the elements using standard voice commands, as Dragon is not able to not pick the fields up by the visible name.

**Current code ref(s):** #countries-input

```
<div class="field"> <label for="countries">
<span class="faux-label">Countries/Regions</span> </label>
  <select id="countries" name="countries[]" class="multiselect"
multiple="multiple">
[...]
</select>
<div class="combo" id="countries-js-multi-buttons">
<div role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-
owns="countries-listbox" class="input-wrapper">
<input aria-activedescendant="countries-input-0" aria-autocomplete="list"
aria-labelledby="countries-label countries-selected" aria-controls="countries- listbox" id="countries-input" class="combo-input" autocomplete="off" type="text"/>
</div>
[...] </div>
</div>
```
**Screen reader user comments:**

> “I examined the combo boxes on the page using the Quick Nav form fields command ‘f’ and the Quick Nav combo boxes command ‘c’, and was disconcerted to find that the 2nd of these, which followed the one announced as ‘Name starts with’, was spoken without a label, meaning that I could not initially understand its purpose. I then reviewed the element in context, and discovered that it related to 'countries/regions', but this took me additional time, and is likely to affect other screen reader users in the same way.
> To ensure easy interaction with this combo box, it should be given a clear label which is spoken in the same way for all navigation methods.
> The issue is present with JAWS, NVDA.”

**Voice activation user comments:**

> “I found I was unable to access the input field labelled ‘Countries/Regions’ by using the labelled name, for example ‘Click Countries/Regions’. The only way I was able to access the input field was by using the command ‘Click Box’. I would have expected to be able to use the labelled name as well.” I tested in: Microsoft Edge

**Solution:**
Please implement this component so that it functions in the anticipated manner for all users. For an example of an accessible autocomplete https://alphagov.github.io/accessible- autocomplete/examples/

Additionally, ensure that labels for fields correctly reference the ID for the corresponding component to ensure that the label is programmatically determinable. The 'id' attribute for the input should match the 'for' attribute of the 'Countries/Regions' label.
The aria-labelledby attribute for this field should be removed, as presently there is no corresponding element for this attribute to reference.

**Comment by deniak** [comment]
> The 'Countries/Region' combobox does not function in the expected manner for users that navigate using the keyboard, as although it is possible to type into the form and cycle through the options which are displayed, it is not possible to make a selection using the space bar as would typically be expected, although pressing enter does work, users would expect to be able to make a selection by pressing Space.

For simple select box, I'm getting different behavior with the space key between Firefox and Chrome. In the latter, hitting space does select the selected element but in Firefox, it does nothing.
However, the [autocomplete examples](https://alphagov.github.io/accessible-autocomplete/examples/) do work as described above even with Firefox.
I couldn't find any document confirming what is expected with the space key but to ensure consistency, I'll see if I can reproduce what autocomplete does.

> In addition to this the combobox presents as being unlabelled to screen reader users due to the labels for the input not correctly referencing the ID of the corresponding component; an aria-labelledby attribute has been assigned as a means of providing a label but this does not reference a corresponding id. As the input is effectively orphaned from its label, it will be difficult for screen reader users to determine its purpose.

I see the problem. I'll submit a fix for the label.
